### DISCUSSION TITLE
How to build from source for iOS

### DISCUSSION THREAD

**Discussion by billyjacoby** [callstack/react-native-visionos#118] [Q&A]
Does anyone have any docs or examples of how to build this repo - or React Native in general - from source for iOS?

It seems there's tons of docs on how to do this for Android but I haven't been able to find anything for how to do so for iOS.

**Comment by okwasniewski** [comment]
Hey, 

If you want to build this project from the source, you can follow the guide described here: 

https://github.com/callstack/react-native-visionos/blob/main/packages/rn-tester/README.md#running-on-ios

You should be able to run it on visionOS simulator

**Reply by billyjacoby** [reply]
I was actually able to get this all setup and running properly yesterday and successfully tested some non-merged features. 

Is there a guide on how to translate the rn-tester app to a more actual react native app? I feel like I have to e missing something simple here. It was also pretty late when I finally got this working last night.

**Reply by okwasniewski** [reply]
You could theoretically do a release to a local npm registry and use it in your app (we are using verdaccio) for this. Release process is described in the README: https://github.com/callstack/react-native-visionos?tab=readme-ov-file#release-process

**Reply by billyjacoby** [reply]
Okay that's helpful, let me try that and I'll report back here with how that all goes. 

Thanks!

**Comment by Saadnajmi** [comment]
Echoing the answer above. As far as I know, there isn't a way with cocoapods (what React Native on Apple platforms uses) to build from source, short of hooking it up to _some_ test app (like RNTester), and then copying the static libraries / frameworks (depending on how you built it) from the Xcode Derived Data folder. To help with this, you can run a local `xcodebuild <args> --derivedDataPath <myFolder>` cli command and pass your own path to DerivedData so you aren't hunting around in Xcode's build artifact directories. This is what I've ended up doing in most places if that's helpful.
